rfcs/20200612-stream-executor-c-api.md

```diff
@@ -308,26 +308,71 @@ typedef struct SE_StreamExecutor {
 
 #define SE_STREAMEXECUTOR_STRUCT_SIZE TF_OFFSET_OF_END(SE_StreamExecutor, host_callback)
 
-TF_CAPI_EXPORT SE_Platform* SE_NewPlatform(
-     SE_PlatformId* id,
-     int32_t visible_device_count,
-     SE_Device* (*create_device)(SE_Options* options, TF_Status* status),
-     void (*destroy_device)(SE_Device* device),
-     SE_StreamExecutor* (*create_stream_executor)(TF_Status* status),
-     void (*destroy_stream_executor)(SE_StreamExecutor* stream_executor);
-);
+typedef struct SE_PlatformParams {
+  size_t struct_size;
+  SE_PlatformId* id;
```

Comment:
Please add a "void* ext" for extensibility, thanks!

Reply:
I added `void* data`. I can rename it to `void* ext` if thats better.

Reasoning is: This struct will be populated by plugin. The only reason a plugin would extend it with a `void*` is to add some additional data that it wants to access later.

Also, this struct is now wrapped with `SE_PlatformRegistrationParams` which has most its fields populated by TF and has `void* ext`.

Reply:
+1 `void* data` looks better!